### PR TITLE
[core, react, vue]: Feature - Connect all previously connected wallets

### PR DESCRIPTION
**Pull request by Adamj1232**
### Description
Adds option to reconnect all previously connected wallets as an option.

Wallets will be reconnected in the order they were previously held within the array.

A new format is introduced within the PR for local storage saving of the wallet labels with an array of strings being used instead of a string. This will be the case regardless if the dev elects usage of `autoConnectAllPreviousWallet` or `autoConnectLastWallet`

```ts
type ConnectModalOptions = {
  ...
  /** 
   * If set to true, all previously connected wallets will store in 
   * local storage. Then on init, onboard will try to reconnect to 
   * each wallet with no modals displayed
   */
  autoConnectAllPreviousWallet?: boolean
}
```

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -227,20 +227,59 @@ function init(options: InitOptions): OnboardAPI {
...
 
-    lastConnectedWallet &&
+    // Handle for legacy single wallet approach
+    if (lastConnectedWallets && typeof lastConnectedWallets === 'string') {
       API.connectWallet({
-        autoSelect: { label: lastConnectedWallet, disableModals: true }
+        autoSelect: { label: lastConnectedWallets, disableModals: true }
       })
+    }
+    if (lastConnectedWallets && Array.isArray(lastConnectedWallets)) {
```
Comment (aaronbarnardsound) [inline packages/core/src/index.ts:244]:
```suggestion
    if (lastConnectedWallets && Array.isArray(lastConnectedWallets && lastConnectedWallets.length)) {
```

Do we need to ensure there are wallets in the array here?

**Review by aaronbarnardsound — APPROVED** [review]
Just one question around checking the list lenght

File: packages/core/src/index.ts (hunk 1 above)
Comment (Adamj1232) [reply packages/core/src/index.ts]: Ahh good call! Added!!

**Comment by Adamj1232** [comment]
> When switching to this branch with a previously connected wallet in the demo project, I got this error: Screenshot 2023-03-23 at 5 00 01 PM
> 
> I can only load the demo app after I clear my local storage and reload.
> 
> Once I refreshed the page- I was able to test and worked as expected- reconnecting all wallets I had connected.

Ahh yep, forgot to add a try block to one area, should be fixed now for backwards compatibility throughout. Thanks for catching that!!

**Pull request merged**